framework/include/base/MooseFunctor.h

```diff
@@ -63,14 +43,18 @@ class Functor : public FunctorBase
    * - a limiter which defines how the functor evaluated on either side of the face should be\n
    *   interpolated to the face
    * - a boolean which states whether the face information element is upwind of the face
+   * - a boolean indicating whether to perform skew correction
    * - a pair of subdomain IDs. These do not always correspond to the face info element subdomain\n
    *   ID and face info neighbor subdomain ID. For instance if a flux kernel is operating at a\n
    *   subdomain boundary on which the kernel is defined on one side but not the other, the\n
    *   passed-in subdomain IDs will both correspond to the subdomain ID that the flux kernel is\n
    *   defined on
    */
-  using FaceArg = std::
-      tuple<const FaceInfo *, Moose::FV::LimiterType, bool, std::pair<SubdomainID, SubdomainID>>;
+  using FaceArg = std::tuple<const FaceInfo *,
```

Comment:
It enables a user to write extremely generic properties like this:
```c++
  addFunctorProperty<ADReal>("total_viscosity", [this](const auto & r, const auto & t) -> ADReal {
    constexpr Real offset = 1e-15; // prevents explosion of sqrt(x) derivative to infinity

    const auto grad_u = _u_vel.gradient(r, t);

    ADReal symmetric_strain_tensor_norm = 2.0 * Utility::pow<2>(grad_u(0));
    if (_mesh_dimension >= 2)
    {
      const auto grad_v = _v_vel->gradient(r, t);

      symmetric_strain_tensor_norm +=
          2.0 * Utility::pow<2>(grad_v(1)) + Utility::pow<2>(grad_v(0) + grad_u(1));
      if (_mesh_dimension >= 3)
      {
        const auto grad_w = _w_vel->gradient(r, t);

        symmetric_strain_tensor_norm += 2.0 * Utility::pow<2>(grad_w(2)) +
                                        Utility::pow<2>(grad_u(2) + grad_w(0)) +
                                        Utility::pow<2>(grad_v(2) + grad_w(1));
      }
    }
    symmetric_strain_tensor_norm = std::sqrt(symmetric_strain_tensor_norm + offset);

    // Return the sum of turbulent viscosity and dynamic viscosity
    return _mu(r, t) +
           _rho(r, t) * symmetric_strain_tensor_norm * Utility::pow<2>(_mixing_len(r, t));
  });
```
That single definition will work if you're evaluating at an element centroid or at a face or at a quadrature point all because the "spatial argument" is always a single argument. E.g. it's all packed into one. If you wanted the same property definition to work without packing ... you just couldn't do it. You'd have to reimplement that same property 'x' number of times where the 'x' is the number of different types of spatial evaluations you want to perform.

Reply:
Got it, thanks!